server/src/main/java/org/opensearch/search/query/QueryPhaseExtension.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.query;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.search.internal.SearchContext;
+
+/**
+ * Extension point interface that allows plugins to hook into the query phase
+ * before and after score collection. This enables custom CollectorManager
+ * implementations and score processing for advanced search features like
+ * hybrid queries and neural search.
+ *
+ * @opensearch.api
+ */
+@PublicApi(since = "3.2.0")
```

Comment:
How confident are you that this is the right signature of these extension points? We will commit to keeping any `@PublicApi` stable for at least the rest of the 3.x line. If you want a little leeway, you can use `@ExperimentalApi` to signal to any components that integrate with this that it is not guaranteed stable. I'd usually recommend marking it experimental until you've got at least one proven integration.